src/ai/defaultai_seafaring.cc

```diff
@@ -127,46 +127,27 @@ bool DefaultAI::marine_main_decisions(const Time& gametime) {
 	uint16_t ports_count = 0;
 	uint16_t shipyards_count = 0;
 	uint16_t expeditions_in_prep = 0;
+	uint16_t expeditions_ready = 0;
 	uint16_t expeditions_in_progress = 0;
 	bool idle_shipyard_stocked = false;
 
-	// goes over all warehouses (these includes ports)
-	for (const WarehouseSiteObserver& wh_obs : warehousesites) {
-		if (wh_obs.bo->is(BuildingAttribute::kPort)) {
-			++ports_count;
-			if (const Widelands::PortDock* pd = wh_obs.site->get_portdock()) {
-				if (pd->expedition_started()) {
-					++expeditions_in_prep;
-				}
-			}
-		}
-	}
-
 	// goes over productionsites and gets status of shipyards
 	for (const ProductionSiteObserver& ps_obs : productionsites) {
 		if (ps_obs.bo->is(BuildingAttribute::kShipyard)) {
 			++shipyards_count;
 
-			// In very rare situation, we might have non-seafaring map but the shipyard is working
-			if (!map_allows_seafaring_ && !ps_obs.site->is_stopped()) {
+			// In very rare situation, we might have non-seafaring map but there is a shipyard
+			if (!map_allows_seafaring_) {
 				verb_log_dbg_time(
 				   game().get_gametime(),
-				   "  %1d: we have working shipyard in a non seafaring ecoomy, stopping it...\n",
+				   "  %1d: we have a shipyard in a non seafaring economy, dismantling it...\n",
 				   player_number());
-				game().send_player_start_stop_building(*ps_obs.site);
-				return false;
-			}
-
-			// counting stocks
```

Comment:
this was moved down into the shipyard handling  to have all the shipyard decisions in one place